src/arena.c

```diff
@@ -803,11 +816,15 @@ arena_destroy(tsd_t *tsd, arena_t *arena) {
 static edata_t *
 arena_slab_alloc(tsdn_t *tsdn, arena_t *arena, szind_t binind, unsigned binshard,
     const bin_info_t *bin_info) {
+	bool deferred_work_generated;
 	witness_assert_depth_to_rank(tsdn_witness_tsdp_get(tsdn),
 	    WITNESS_RANK_CORE, 0);
 
 	edata_t *slab = pa_alloc(tsdn, &arena->pa_shard, bin_info->slab_size,
-	    PAGE, /* slab */ true, /* szind */ binind, /* zero */ false);
+	    PAGE, /* slab */ true, /* szind */ binind, /* zero */ false,
+	    &deferred_work_generated);
+
+	assert(deferred_work_generated == false);
```

Comment:
Is this assert right? It seems like a slab allocation could trigger a (delayed) hugification, so that we would in fact generate deferred work. No?

(Also, just stylistically, for booleans we'd usually assert the bool computation directly rather than its value; i.e. `assert(!deferred_work_generated)` instead of `assert(deferred_work_generated == false)`).

Reply:
Nice catch. Aside from removing the assertion, I need to call `arena_handle_deferred_work` to wake up from possible indefinite sleep.